src/status_im/ui/screens/discover/events.cljs

```diff
@@ -147,6 +147,12 @@
     (fn [_ _]
       (discoveries/delete :created-at :asc 1000 200))))
 
+(handlers/register-handler-fx
+ :show-discovery
+ (fn [_ [_ tags view-id]]
+   {:dispatch-n [[:set :discover-search-tags tags]
+                 [:navigate-to view-id]]}))
```

Comment:
Can you please refactor the code to not use `:dispatch-n`, but functional composition instead?
```clj
(handlers/register-handler-db
  :show-discovery
  (fn [db [_ tags view-id]]
    (-> db
        (assoc :discover-search-tags tags)
        (navigation/navigate-to view-id))))
```
With `navigation` ns (`status-im.ui.screens.navigation`) required of course. 
The reason is that we are trying to move from imperative, side-effecty code
to something more functional, see the rationale - https://github.com/status-im/ideas/issues/31 (Point 3.)